test/types/index_options.test-d.ts

```diff
@@ -0,0 +1,28 @@
+import { expectType, expectNotType } from 'tsd';
+import { IndexDescription } from '../../src';
+
+// test that all valid index options are allowed in IndexDefinitions
+expectType<IndexDescription>({
+  key: {},
+  background: true,
+  expireAfterSeconds: 2400,
+  name: 'index_1',
+  sparse: true,
+  storageEngine: {},
+  unique: true,
+  version: 1,
+  default_language: 'english',
+  language_override: 'english',
```

Comment:
Note that these are camel case in the spec, too late to update them to conform for 4.0?

Reply:
I think so, we could add support for either one? Snake and camel variations are permitted in each driver but this is just unusual for us. 🤷  It's a small inconsistency we can live with if we want to just leave it alone.

Reply:
I'm fine with leaving as is, it's minor enough that I think it's better to avoid further scope creep in this PR.